packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -928,14 +928,21 @@ const writeCacheFile = (cachePath: string, fileData: string) => {
  * processes attempt to rename to the same target file at the same time.
  * If the target file exists we can be reasonably sure another process has
  * legitimately won a cache write race and ignore the error.
+ * If the target does not exist we do not know if it is because it is still
+ * being written by another process or is being overwritten by another process.
  */
 const cacheWriteErrorSafeToIgnore = (
   e: Error & {code: string},
   cachePath: string,
-) =>
-  process.platform === 'win32' &&
-  e.code === 'EPERM' &&
-  fs.existsSync(cachePath);
+) => {
+  if (process.platform !== 'win32' || e.code !== 'EPERM') {
+    return false;
+  }
+  if (!fs.existsSync(cachePath)) {
+    console.warn('Possible problem writing cache if this occurs many times', e);
```

Comment:
yes, I'm not sure exactly what to do...

There may be people where writing the cache file fails on windows and its down to permissions, so it happens every time.

But there is also the situation where this occasionally fails all the time intermittently.

So my idea was that it would warn constantly on the console for the first situation and not at all for the second situation.

Currently I am using https://github.com/fluffynuts/patch-jest-cache which I believe does not warn - so I'm happy to remove the warning or if it is a concern, make this a option.